server/schedule/operator_controller.go

```diff
@@ -179,11 +200,13 @@ func (oc *OperatorController) pollNeedDispatchRegion() (r *core.RegionInfo, next
 	r = oc.cluster.GetRegion(regionID)
 	if r == nil {
 		_ = oc.removeOperatorLocked(op)
-		log.Debug("remove operator because region disappeared",
-			zap.Uint64("region-id", op.RegionID()),
-			zap.Stringer("operator", op))
-		operatorCounter.WithLabelValues(op.Desc(), "disappear").Inc()
-		oc.opRecords.Put(op, pdpb.OperatorStatus_CANCEL)
+		if op.Cancel() {
```

Comment:
Will it be false if `Cancel` is concurrency called as I mention before? I think we can let be idempotent.

Reply:
Yes, it will, and it also *should* be. Only the thread that really changed the status should get a `true`, or multiple thread will log this operator.